### PR TITLE
NEGOTIATE authentication error: No valid credentials provided (Mechanism level: No valid credentials provided (Mechanism level: Failed to find any Kerberos tgt))

### PR DESCRIPTION
**Pull request by rlambrechts**
Fix issue for #144 and issue #160 (duplicate)

Confirmed to be working: https://github.com/OfficeDev/ews-java-api/issues/144

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@rlambrechts__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        This seems like a small (but important) contribution, so no contribution license agreement is required at this point. Real humans will now evaluate your PR.
    </span>

TTYL, AZPRBOT;

File: src/main/java/microsoft/exchange/webservices/data/HttpClientWebRequest.java [hunk 1]
```diff
@@ -127,6 +138,14 @@ public void prepareConnection() throws EWSHttpException {
...
       rcBuilder.setRedirectsEnabled(isAllowAutoRedirect());
       rcBuilder.setSocketTimeout(getTimeout());
+	  
+	  // fix issue #144 + #160: if we used NTCredentials from above: these are NT credentials
+      if (getUserName() != null) {
+    	ArrayList<String> authPrefs = new ArrayList<String>();
+    	authPrefs.add(AuthSchemes.NTLM);
+    	rcBuilder.setTargetPreferredAuthSchemes(authPrefs).build();
+      }
+      //
```
Comment (serious6) [inline src/main/java/microsoft/exchange/webservices/data/HttpClientWebRequest.java]: * should this behavior also be added to `prepareAsyncConnection`? 
* `rcBuilder.build()` is called twice ref. [l146](https://github.com/rlambrechts/ews-java-api/blob/master/src/main/java/microsoft/exchange/webservices/data/HttpClientWebRequest.java#L146) / [l149](https://github.com/rlambrechts/ews-java-api/blob/master/src/main/java/microsoft/exchange/webservices/data/HttpClientWebRequest.java#L149) is this usefull?
* please align the comment in [l142](https://github.com/rlambrechts/ews-java-api/blob/master/src/main/java/microsoft/exchange/webservices/data/HttpClientWebRequest.java#L142)

Comment (rlambrechts) [inline src/main/java/microsoft/exchange/webservices/data/HttpClientWebRequest.java]: - prepareAsyncConnection: I assule so however I haven't verified this.
- build() => fixed.

File: src/main/java/microsoft/exchange/webservices/data/HttpClientWebRequest.java [hunk 2]
```diff
@@ -189,6 +208,13 @@ public void prepareAsyncConnection() throws EWSHttpException {
       rcBuilder.setConnectionRequestTimeout(getTimeout());
       rcBuilder.setConnectTimeout(getTimeout());
       rcBuilder.setSocketTimeout(getTimeout());
+	  
+      // fix issue #144 + #160: if we used NTCredentials from above: these are NT credentials
+      ArrayList<String> authPrefs = new ArrayList<String>();
```
Comment (barisaydinoglu) [inline src/main/java/microsoft/exchange/webservices/data/HttpClientWebRequest.java]: prepareAsyncConnection doesn't have same controls with prepareConnection.
```if (getUserName() != null) ... ```

**Comment by serious6** [comment]
@rlambrechts thanks for the contribution. If you have adressed the above comment and there are no other suggestions we can have this merged. Maybe you could also confirm the behavior for the async request?

**Comment by MikeN123** [comment]
This PR seems to disable Basic auth, which means this library won't work with O365. I'm currently adding a commit to #180 (which conflicts with this PR anyway), to enable both NTLM and Basic auth, on target and proxy.

**Comment by vboctor** [comment]
Thanks @rlambrechts for your contribution.  I've just merged #180 which included the port of this fix by @MikeN123.  Enabling basic auth will keep the Office 365 access working.

**Pull request closed without merging**